module/userProjects/helper.js

```diff
@@ -2280,6 +2325,359 @@ module.exports = class UserProjectsHelper {
         })
     }
 
+    /**
+     * generate project certificate.
+     * @method
+     * @name generateCertificate 
+     * @param {Object} data - project data for certificate creation data.
+     * @returns {JSON} certificate details.
+    */
+
+     static generateCertificate(data) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                //  if eligible key is not there check criteria for validation
+                if ( !data.certificate.eligible ) {
+                    let validateCriteria = await certificateValidationsHelper.criteriaValidation(data)
+                    if ( validateCriteria.success ) {
+                        data.certificate.eligible = true;
+                    } else {
+                        data.certificate.eligible = false;
+                        data.certificate.message = validateCriteria.message
+                    }
+                }
+                let updateObject = {
+                    "$set" : {}
+                };
+                updateObject["$set"]["certificate.eligible"] =  data.certificate.eligible;
+                if ( data.certificate.message && data.certificate.message !=="" ) {
+                    updateObject["$set"]["certificate.message"] =  data.certificate.message;
+                }
+                if ( Object.keys(updateObject["$set"]).length > 0 ) {
+                    await projectQueries.findOneAndUpdate(
+                        {
+                            _id: data._id
+                        },
+                        updateObject
+                    );
+                } 
+                if ( data.certificate.eligible === true && ( data.certificate.transactionId || data.certificate.osid ) ) {
+                    return resolve( { 
```

Comment:
@VISHNUDAS-tunerlabs instead of repeating the same code, just throw and error with message.